processing/python/display_python/surfaces.py

```diff
@@ -0,0 +1,144 @@
+        self.font = font
+    
+    def update_utts(self, utt, cat):
+        utt_cat_surf = UtteranceLine(self.surface.width, self.surface.height, utt, cat, self.font, self.pos_x)
+        self.add_subsurface(self.index, utt_cat_surf)
+        self.index += 1
+               
+        pos_y = 0
+        self.iterate = True
+        surfaces_to_iterate = reversed(list(self.subsurfaces.values()))
+        for utt_line in surfaces_to_iterate:
+            utt_line.set_pos_y(pos_y)
+            if pos_y >= self.surface.height:
+                break 
+            pos_y += utt_line.surface.height
+    
+        if pos_y > self.surface.height:
+            self.subsurfaces.popitem(last=False)  
+             
+class Beat:
+    def __init__(self, s_width, s_height, beatnum, col, font):
+        self.surface = createGraphics(s_width, s_height)
+        self.update_beatnum(beatnum, font, col)
+    
+    def update_beatnum(self, beatnum, font, col):
+        with self.surface.beginDraw():
+            self.surface.background(222)
+            self.surface.textFont(font)
+            self.surface.textSize(100)
+            self.surface.fill(col)
+            self.surface.textAlign(CENTER)
+            self.surface.text(beatnum, self.surface.width/2, self.surface.height * 3/5)
+        
+    def draw(self, surface):
+        with surface.beginDraw():
+            surface.image(self.surface, self.surface.width *4/5, 0) 
+    
+class Parts:
+    def __init__(self, s_width, s_height, current, next, font, col):
+        self.surface = createGraphics(s_width, s_height)
+        self.update_parts(current, next, font, col)
+    
+    def update_parts(self, current, next, font, col):
+        with self.surface.beginDraw():
+            self.surface.background(222)
+            self.surface.textFont(font)
+            self.surface.textSize(20)
+            self.surface.fill(0)
+            self.surface.textAlign(CENTER)
+            self.surface.text("current part:\n" + current, self.surface.width/2, self.surface.height/4)
+            self.surface.text("next part:\n" + next, self.surface.width/2, self.surface.height*3/4)
+    
+    def draw(self, surface):
+        with surface.beginDraw():
+            surface.image(self.surface, 0, 0) 
+    
+    
+class PartArea(SurfaceBase):
+    def __init__(self, name, pos_x, pos_y, s_width, s_height, font):
+        SurfaceBase.__init__(self, name, pos_x, pos_y, s_width, s_height)
+        self.font = font
+    
+    def update_parts(self, current, next, beatnum, change):
+        if change == "True":
+            col = color(250, 0, 20)
+        else: 
+            col = color(10, 250, 0)
+        beat_surf = Beat(self.surface.width/2, self.surface.height, beatnum, col, self.font)
+        current_next_surf = Parts(self.surface.width/2, self.surface.height, current, next, self.font, col)
+        self.add_subsurface("beat", beat_surf)
+        self.add_subsurface("parts", current_next_surf)
+   
+
+def build_areas2(font):
```

Comment:
I don't see an advantage of declaring `AREAs` as a module-attribute. I would just define `areas = {}` here, add the two instances and return it.

Reply:
@staudamm 
es könnte sein, daß es nötig ist, weil font nur im setup() deklariert werden kann und über build_areas(font) auf diese Seite "geschickt" werden kann. Ich probier aber mal.
Sonst hast du überall anders  total recht (und ich entsprechend unrecht ;-))

Reply:
hello @staudamm 
ich habe die build_areas funktion erhalten, damit nicht zuviel kram in setup() steht. dafür ist sie jetzt auf dem Hauptsketch window, was ja wohl sinn macht, gell?
also dann merge ich des jetzt